### ISSUE TITLE
E2533 - has not worked since v1.3.4

### ISSUE THREAD

**Issue by whoDoneItAgain** [aws-cloudformation/cfn-lint#4191]
### CloudFormation Lint Version

v1.3.4

### What operating system are you using?

windows

### Describe the bug

E2533 has not reported since 1.3.4. I'm actually using 1.38.0 and discovered it wasn't working there. I rolled back my version until i found it to work again in 1.3.3

### Expected behavior

E2533 should be reporting deprecated lambda runtimes 

### Reproduction template

```
AWSTemplateFormatVersion: '2010-09-09'

Parameters:
  Runtime:
    Description: Lambda Runtime
    Type: String
    Default: python3.13
    AllowedValues:
      - dotnet6
      - dotnet8
      - dotnetcore1.0
      - dotnetcore2.0
      - dotnetcore2.1
      - dotnetcore3.1
      - go1.x
      - java11
      - java17
      - java21
      - java8
      - java8.al2
      - nodejs
      - nodejs10.x
      - nodejs12.x
      - nodejs14.x
      - nodejs16.x
      - nodejs18.x
      - nodejs20.x
      - nodejs22.x
      - nodejs4.3
      - nodejs4.3-edge
      - nodejs6.10
      - nodejs8.10
      - provided
      - provided.al2
      - provided.al2023
      - python2.7
      - python3.10
      - python3.11
      - python3.12
      - python3.13
      - python3.6
      - python3.7
      - python3.8
      - python3.9
      - ruby2.5
      - ruby2.7
      - ruby3.2
      - ruby3.3
      - ruby3.4

Resources:
  Lambda:
    Type: AWS::Lambda::Function
    Properties:
      Architectures:
        - arm64
      Code:
        ZipFile: |
          def handler(event, context):
              LOGGER.info("## EVENT")
              LOGGER.info(event)
      Handler: index.handler
      Role: arn:aws:iam::123456789012:role/test
      Runtime: !Ref Runtime
```

**Comment by kddejong** [comment]
Broke when resolving: https://github.com/aws-cloudformation/cfn-lint/issues/3380
